exchange/bidder_test.go

```diff
@@ -3347,3 +3348,61 @@ func TestDoRequestImplWithTmaxTimeout(t *testing.T) {
 		test.assertFn(httpCallInfo.err)
 	}
 }
+
+func BenchmarkCompressToGZIPNotOptimized(b *testing.B) {
+	// Setup the mock server
+	respBody := "{\"bid\":false}"
+	respStatus := 200
+	server := httptest.NewServer(mockHandler(respStatus, "getBody", respBody))
+	defer server.Close()
+
+	// Prepare the request data
+	req := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     server.URL,
+		Body:    []byte("{\"key\":\"val\"}"),
+		Headers: http.Header{},
+	}
+
+	// Run the benchmark
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		setRequestBodyNotOptimized(req.Body)
+	}
+}
+
+func BenchmarkCompressToGZIPOptimized(b *testing.B) {
+	// Setup the mock server
+	respBody := "{\"bid\":false}"
+	respStatus := 200
+	server := httptest.NewServer(mockHandler(respStatus, "getBody", respBody))
+	defer server.Close()
+
+	// Prepare the request data
+	req := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     server.URL,
+		Body:    []byte("{\"key\":\"val\"}"),
+		Headers: http.Header{},
+	}
+
+	// Run the benchmark
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		setRequestBody(req, "GZIP")
+	}
+}
+
+func setRequestBodyNotOptimized(requestBody []byte) ([]byte, error) {
+	var b bytes.Buffer
+	w := gzip.NewWriter(&b)
+	_, err := w.Write([]byte(requestBody))
+	if err != nil {
+		return nil, err
+	}
+	err = w.Close()
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
```

Comment:
I recommend deleting `BenchmarkCompressToGZIPNotOptimized` and `setRequestBodyNotOptimized`. In the future, you would run `BenchmarkCompressToGZIPOptimized` before and after the changes.